.github/workflows/build.yml

```diff
@@ -1,127 +1,139 @@
-                  node-version: ${{ matrix.node }}
-
-            - uses: pnpm/action-setup@v2.0.1
-              with:
-                  version: 7.6.0
-
-            - name: Build @hashgraph/sdk
-              run: task build
-
-    test:
-        runs-on: ubuntu-latest
-        strategy:
-            matrix:
-                node: ["16"]
-
-        steps:
-            - uses: actions/checkout@v2
-              with:
-                  submodules: recursive
-
-            - name: Install Task
-              uses: arduino/setup-task@v1
-              with:
-                  version: 3.7.0
-
-            - name: "Create env file"
-              run: |
-                  touch .env
-                  echo OPERATOR_KEY="302e020100300506032b65700422042091132178e72057a1d7528025956fe39b0b847f200ab59b2fdd367017f3087137" >> .env
-                  echo OPERATOR_ID="0.0.2" >> .env
-                  echo HEDERA_NETWORK="local-node" >> .env
-                  cat .env
-
-            - name: Cache pnpm modules
-              uses: actions/cache@v2
-              env:
-                  cache-name: cache-pnpm-modules
-              with:
-                  path: ~/.pnpm-store
-                  key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ matrix.node-version }}-${{ hashFiles('**/package.json') }}
-                  restore-keys: |
-                      ${{ runner.os }}-build-${{ env.cache-name }}-${{ matrix.node-version }}-
-
-            - uses: actions/setup-node@v2
-              with:
-                  node-version: ${{ matrix.node }}
-
-            - uses: pnpm/action-setup@v2.0.1
-              with:
-                  version: 7.6.0
-
-            - name: Build @hashgraph/sdk
-              run: task build
-
-            - name: Start the local node
-              run: npx @hashgraph/hedera-local start -d --network local
-
-            - name: Run Hedera SDK Integration Tests Codecov
-              run: task test:integration:codecov
-
-            - name: Stop the local node
-              run: npx @hashgraph/hedera-local stop
-
-            - name: Build @hashgraph/cryptography
-              working-directory: packages/cryptography
-              run: task build
-
-            - name: Uint Test @hashgraph/cryptography
-              working-directory: packages/cryptography
-              run: task test:unit
-
-            - name: Codecov @hashgraph/cryptography
-              working-directory: packages/cryptography
-              run: task test:unit:codecov
-
-            - name: Unit Test @hashgraph/sdk
-              run: task test:unit
-
-            - name: Codecov @hashgraph/sdk
-              run: task test:unit:codecov
+  build:
+    name: Build using Node ${{ matrix.node }}
+    runs-on: [self-hosted, Linux, medium, ephemeral]
+    strategy:
+      matrix:
+        node: [ "16", "18" ]
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

Comment:
Using a version tag like `v4` or `v4.1.1` is less secure because the tags can be moved to a malicious commit in the upstream repository and lead to us being compromised. Using the SHA of the commit guarantees we either get the exact code we trust or the job fails. 

This is a best practice to help mitigate supply chain attacks.